src/BatteryStateOfChargeService.h

```diff
@@ -17,6 +17,15 @@ class BatteryStateOfChargeService : public I_BatteryStateOfChargeService
 private:
     // Convention is to have an _ after private members
     const double initialStateOfChargePercent_;
-
+    double objectCurrent;
+    double inCurrent;
+    double changeTime;
+    QTime initialTime;
```

Comment:
This variable is being updated with every new `initialTime` so the name is not really accurate, "initial" implies that this variable is for the very very first time.